google/cloud/storage/internal/curl_client.cc

```diff
@@ -225,6 +225,18 @@ StatusOr<ResumableUploadResponse> CurlClient::QueryResumableUpload(
   return AsStatus(*response);
 }
 
+StatusOr<std::unique_ptr<ResumableUploadSession>>
+CurlClient::FullyRestoreResumableSession(ResumableUploadRequest const& request,
+                                         std::string const& session_id) {
+  auto session = absl::make_unique<CurlResumableUploadSession>(
+      shared_from_this(), request, session_id);
+  auto response = session->ResetSession();
+  if (response.status().ok()) {
```

Comment:
Nit: Do we prefer to say this over just `if (response)`?

Reply:
Yes, we do. Thanks.